src/main/java/io/confluent/connect/jdbc/sink/dialect/DbDialect.java

```diff
@@ -157,9 +157,12 @@ static void formatColumnValue(StringBuilder builder, Schema.Type type, Object va
   }
 
   protected String getSqlType(String schemaName, Schema.Type type) {
-    final String sqlType;
-    if (schemaName != null && logicalNameToSqlTypeMap.containsKey(schemaName)) {
+    String sqlType;
+    if (schemaName != null) {
       sqlType = logicalNameToSqlTypeMap.get(schemaName);
+      if (sqlType == null) {
+        sqlType = schemaTypeToSqlTypeMap.get(type);
+      }
     } else {
```

Comment:
Ah what I was suggesting before was that we could have the `if (sqlType == null)` here instead of an else branch, so don't need to repeat code for the `schemaTypeToSqlTypeMap` lookup